asdf/asdf.py

```diff
@@ -1157,10 +1157,13 @@ def update(self, all_array_storage=None, all_array_compression='input',
         if version is not None:
             self.version = version
 
+        if all_array_storage in ['internal', 'external']:
```

Comment:
Should `all_array_storage == "inline"` also override the `auto_inline` parameter?

I'd recommend moving this code inside the `_pre_write` function, since that's a natural place for checks that are common to `update` and `write_to`.